src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs

```diff
@@ -48,14 +48,14 @@ public static ParseUserDefinitionResult Parse(string script, ParserOptions parse
             return TexlParser.ParseUserDefinitionScript(script, parserOptions);
         }
 
-        public static bool ProcessUserDefinitions(string script, INameResolver globalNameResolver, IBinderGlue documentBinderGlue, BindingConfig bindingConfig, ParserOptions parserOptions, out UserDefinitionResult userDefinitionResult, Features features = null)
+        public static bool ProcessUserDefinitions(string script, INameResolver globalNameResolver, IBinderGlue documentBinderGlue, BindingConfig bindingConfig, ParserOptions parserOptions, out UserDefinitionResult userDefinitionResult, Features features = null, bool shouldBindBody = false)
```

Comment:
What's the point of this flag?
 Also, this UDF codepath is mostly new code, there should be documentation comments for most of these methods/parameters.

Reply:
When would it make sense to not bind, when calling this?
I think this pattern is somewhat off, would be good to consider reworking to similar to Engine's handling of expression:

Parse(string) -> ParseResult
Check(ParseResult) -> CheckResult (does binding)
Eval(CheckResult) -> ....

Reply:
The single method doing everything with a bunch of params feels odd

Reply:
For UDFs, we bind the body on declaration, to verify if provided return type matches with actual return type. This may be useful for other hosts, but for power apps this results in duplicate binding.

Reply:
Added comments & cleaned up the API